source/lesson-2-ruby-and-the-command-line.html.md.erb

```diff
@@ -0,0 +1,722 @@
+
+### Running ruby interactively with `irb`
+
+There are two ways to run ruby from the command line:
+
+* Interactively with `irb` (**I**nteractive **R****u**by)
+* Running a file with `ruby` itself
+
+To start with, let's look at `irb`. We'll get on to how to run a file later.
+
+`irb` is what's known as a REPL (**R**ead **E**valuate **P**rint **L**oop) -
+it lets us write ruby expressions, and quickly see what they evaluate to.
+
+Start `irb` by running:
+
+```shell
+$ irb
+```
+
+Your command line prompt will change from `$ ` to `irb(main):001:0> `.
+
+When you want to leave irb and go back to your command prompt, just type `exit` and press enter.
+
+### Numbers
+
+We can use ruby as a kind of calculator. Try typing `1 + 1` into irb and
+pressing enter. Do you get the right answer?
+
+In ruby, the `+` operator adds numbers together. Other operators include:
+
+* `-` - subtract
+* `*` - multiply
+* `/` - divide
+* `**` - raise to the power of
+
+You can also use brackets (`()`) to group expressions.
+
+#### Task 2: Maths challenge
+
+Use `irb` to work out the answer to "191 multiplied by 7".
+
+<details>
+<summary>Answer</summary>
+
+```
+irb(main):016:0> 191 * 7
+=> 1337
+```
+
+</details>
+
+### Strings
+
+In the real world strings tie things up. Programming strings have *nothing*
+to do with real-world strings.
+
+Programming strings are used to store collections of letters and numbers.
+That could be a single letter like `"a"`, a word like `"hi"`, or a sentence like
+`"Hello my friends."`.
+
+A Ruby string is written as a quote (`"`) followed by some letters, numbers,
+or symbols and ended with a closing quote (`"`). The shortest possible string
+is called the empty string: `""`. It’s not uncommon for a single string to
+contain paragraphs or even pages of text.
+
+If you type a string into `irb` it will echo it back at you:
+
+```
+irb(main):017:0> "Hello IRB"
+=> "Hello IRB"
+```
+
+#### Getting things wrong in IRB
+
+Just like in the command line, if you give `irb` a command it doesn't
+understand it will show you an error message. Again, don't panic! Everything
+is fine.
+
+`irb` will do its best to explain why it didn't understand your ruby, but
+sometimes the error messages can be hard for a beginner to understand.
+
+For example:
+
+```
+irb(main):018:0> "Hello IRB" + 7
+TypeError: no implicit conversion of Fixnum into String
+        from (irb):1:in `+'
+        from (irb):1
+        from /usr/bin/irb:11:in `<main>'
+```
+
+Here, `irb` is telling us that you're not allowed to add strings and numbers
+together in ruby.
+
+### Variables
+
+Programming is all about creating abstractions, and in order to create an
+abstraction we must be able to assign names to things. Variables are a way of
+creating a name for a piece of data.
+
+Creating and using variables in ruby is super simple, it's just `name` `=`
+`expression`. Some examples:
```

Comment:
can we go with 'value'? That resonated with me more..

Reply:
Yep, done.